### DISCUSSION TITLE
Should included Nerd Font have all character cell widths set to 1 cell? 0.36 onwards

### DISCUSSION THREAD

**Discussion by benfrain** [kovidgoyal/kitty#7774] [Q&A] [answered]
Really great to see a nerd font included by default in Kitty now from 0.36 onwards. However, I still need to use:

```
narrow_symbols U+23FB-U+23FE,U+2665,U+26A1,U+2B58,U+E000-U+E00A,U+E0A0-U+E0A3,U+E0B0-U+E0C8,U+E0CA,U+E0CC-U+E0D2,U+E0D4,U+E200-U+E2A9,U+E300-U+E3E3,U+E5FA-U+E634,U+E700-U+E7C5,U+EA60-U+EBEB,U+F000-U+F2E0,U+F300-U+F32F,U+F400-U+F4A9,U+FC1F,U+F918,U+F500-U+F8FF,U+DB82-U+DDDE 1
```

To get the symbols displaying the correct width in Neovim menus. More on that approach here: https://benfrain.com/how-to-get-equal-size-icons-in-the-cmp-completion-menu-of-neovim-with-kitty-terminal/

I wonder whether the included default Nerd Font should/could always have the cells set to 1? Should that be the default behaviour?

Would there be occasions when this would not be desirable?

**Comment by kovidgoyal** [comment, marked as answer]
The default is to draw the glyphs at their natural width, if there is
space available. The default should be the best appearance, which is the natural width.
Squashing a square icon into a rectangular single cell looks horrible.

If you want to restrict them to one cell then you need to configure it explicitly as you have done.
For your particular issue, the correct fix is to use icons in that menu
that all are square or dont follow them with a space so kitty will not
expand them by default.

**Reply by benfrain** [reply]
Interestingly, it seems that there is an issue when using `pumblend` option in Neovim that causes an extra glyph in these menus: https://github.com/onsails/lspkind.nvim/issues/55#issuecomment-1294996794

So, another option here is to disable the transparency in the menu and then the `narrow_symbols` are also unneeded.

**Comment by kovidgoyal** [comment]
Well, the long term solution for that is for me to get around to
implementing native popup windows, then editors can avoid hacks like
pumblend.
